DiverseMarket.UI/Pages/Company/Product/AddNewProductPage.Designer.cs

```diff
@@ -0,0 +1,387 @@
+            //categoryLabel.Location = new Point(354, categoryComboBox.Top - spacing);
+            categoryLabel.Location = new Point(354, categoryTextBox.Top - spacing);
+            categoryLabel.ForeColor = Color.White;
+            categoryLabel.Font = new Font("Ubuntu", 12);
+            this.Controls.Add(categoryLabel);
+
+            Label quantityLabel = new Label();
+            quantityLabel.Text = "Quantidade";
+            quantityLabel.Location = new Point(354, quantityTextBox.Top - spacing);
+            quantityLabel.ForeColor = Color.White;
+            quantityLabel.Font = new Font("Ubuntu", 12);
+            this.Controls.Add(quantityLabel);
+
+            Label priceLabel = new Label();
+            priceLabel.Text = "Preço";
+            priceLabel.Location = new Point(354, priceTextBox.Top - spacing);
+            priceLabel.ForeColor = Color.White;
+            priceLabel.Font = new Font("Ubuntu", 12);
+            this.Controls.Add(priceLabel);
+        }
+
+        #endregion
+
+        #region Logo
+        private void InitLogo()
+        {
+            this.Icon = new Icon(@"Resources\icon.ico");
+
+            Logo logo = new Logo();
+            logo.Location = new Point(820, 77);
+            logo.Width = 192;
+            logo.Height = 22;
+
+            this.Controls.Add(logo);
+        }
+        #endregion
+
+        #region Buttons
+        private void InitButtons()
+        {
+
+            int spacing = 50;
+
+            #region HomePage Button
+            this.homepageButton = new System.Windows.Forms.Button();
+            this.homepageButton.Location = new Point(37, 67);
+            this.homepageButton.Size = new Size(79, 71);
+            this.homepageButton.FlatStyle = FlatStyle.Flat;
+            this.homepageButton.FlatAppearance.BorderSize = 0;
+            this.homepageButton.Cursor = Cursors.Hand;
+            this.homepageButton.Image = Image.FromFile(@"Resources\logo-reduzida.png");
+            this.homepageButton.BackgroundImageLayout = ImageLayout.Zoom;
+            this.homepageButton.Click += new EventHandler((object sender, EventArgs e) =>
+            {
+                this.Hide();
+                new HomePageCompany(this._userId).Show();
+            });
+
+            this.Controls.Add(homepageButton);
+
+            #endregion
+
+            this.addNewOfferButton = new RoundedButton("Adicionar Oferta", 150, 57, Colors.SecondaryButton, 32);
+            this.addNewOfferButton.Location = new System.Drawing.Point(557, priceTextBox.Bottom + spacing);
+            this.addNewOfferButton.MouseEnter += new EventHandler((object sender, EventArgs e) =>
+            {
+                this.addNewOfferButton.Cursor = Cursors.Hand;
+            });
+            this.addNewOfferButton.Click += new EventHandler((object sender, EventArgs e) =>
+            {
+                this.Hide();
+                new AddSpecificOfferPage(this._userId).Show();
```

Comment:
Dá pra melhorar, mas podemos chamar mais ou menos assim: 

```suggestion
                var newProductOfferInsertDTO = new ProductOfferInsertDTO(
            this._userId,
            decimal.Parse(this.priceTextBox.TextBox.Text),
            long.Parse(this.quantityTextBox.TextBox.Text),
            this.nameTextBox.TextBox.Text,
            this.categoryComboBox.SelectedItem.ToString(),
            this.descriptionTextBox.TextBox.Text
        );


ProductService.InsertNewProductWithProductOffer(newProductOfferInsertDTO);

bool wasUpdateSuccessful = ProductService.InsertNewProductWithProductOffer(newProductOfferInsertDTO);

if (wasUpdateSuccessful)
{
    MessageBoxUtils.ShowMessageBox("Oferta inserida com sucesso!", MessageBoxIcon.Information);
}
else
{
    MessageBoxUtils.ShowMessageBox("Falha ao inserir a oferta. Tente novamente.", MessageBoxIcon.Error);
}
```